### PR TITLE
Upgrade navigation

### PR DESCRIPTION
**Pull request by lieblichaaron**
## ℹ️ Change description:
Nothing here

<sub>[templates: [bug](?expand=1&template=bug.md), [feature](?expand=1&template=feature.md)]</sub>

### CODE REVIEW

File: locomotion-sdk/package.json [hunk 1]
```diff
@@ -40,17 +40,22 @@
...
     "@stripe/stripe-react-native": "^0.8.0",
-    "react-navigation": "^3.13.0"
+    "react-native-safe-area-context": "^4.2.5",
+    "react-native-screens": "^3.13.1",
+    "@react-navigation/native": "^6.0.10",
+    "@react-navigation/drawer": "^6.3.0",
+    "@react-navigation/native-stack": "^6.6.2"
   },
   "dependencies": {
+    "@react-navigation/drawer": "^6.3.0",
```
Comment (OfryL) [inline locomotion-sdk/package.json:53]: need to verify its not peer dep
@oriefrati

Comment (lieblichaaron) [reply locomotion-sdk/package.json:53]: it is, i added them to dependancies also so i could work on the code with types

File: locomotion-sdk/src/Components/Drawer/index.js [hunk 2]
```diff
@@ -23,10 +26,9 @@ export const DrawerLabel = (props) => {
 };
 
 export const DrawerContentComponent = (props) => {
-  const { navigation, screenProps } = props;
-
+  const navigation = useNavigation()
```
Comment (avivbarniv) [inline locomotion-sdk/src/Components/Drawer/index.js]: remove or use this var in the closeDrawer function

File: examples/client/Locomotion/android/app/src/main/java/com/locomotion/MainApplication.java [hunk 3]
```diff
@@ -31,14 +33,19 @@ protected List<ReactPackage> getPackages() {
           // Packages that cannot be autolinked yet can be added manually here, for example:
           // packages.add(new MyReactNativePackage());
           packages.add(new LinearGradientPackage());
-          packages.add(new RNMixpanel());
+          // packages.add(new RNMixpanel());
```
Comment (oriefrati) [inline examples/client/Locomotion/android/app/src/main/java/com/locomotion/MainApplication.java]: not needed anymore?

File: locomotion-sdk/src/pages/index.js [hunk 4]
```diff
@@ -1,109 +1,22 @@
-import React from 'react';
-import { Dimensions, View } from 'react-native';
-import {
-  createSwitchNavigator, createStackNavigator, createAppContainer, createDrawerNavigator,
-} from 'react-navigation';
-import i18n from '../I18n';
-
+import * as React from 'react';
```
Comment (avivbarniv) [inline locomotion-sdk/src/pages/index.js]: Why do you need to import it like this?

File: examples/client/Locomotion/package.json [hunk 5]
```diff
@@ -10,14 +10,17 @@
...
     "@mapbox/polyline": "^1.1.0",
     "@react-native-async-storage/async-storage": "^1.17.3",
     "@react-native-community/datetimepicker": "^6.1.3",
     "@react-native-community/geolocation": "^2.0.2",
     "@react-native-firebase/app": "^14.9.1",
     "@react-native-firebase/crashlytics": "^14.9.1",
     "@react-native-picker/picker": "^2.4.1",
+    "@react-navigation/drawer": "^6.3.0",
+    "@react-navigation/native": "^6.0.10",
+    "@react-navigation/native-stack": "^6.6.2",
```
Comment (oriefrati) [inline examples/client/Locomotion/package.json]: is it native package?

File: locomotion-sdk/package.json (hunk 1 above)
Comment (oriefrati) [reply locomotion-sdk/package.json:53]: are they native dependancies?

File: locomotion-sdk/src/LocomotionRouter.js [hunk 6]
```diff
@@ -3,12 +3,12 @@ import Config from 'react-native-config';
...
 
 import { MainProvider } from './context/main';
-import Router, { MainRouter } from './pages';
-import NavigationService from './services/navigation';
+import MainRouter from './pages';
 
 import RidePopups from './popups/RidePopups';
 import SettingsContext from './context/settings'
 import PaymentsContext from './context/payments'
+import 'react-native-gesture-handler';
```
Comment (oriefrati) [inline locomotion-sdk/src/LocomotionRouter.js]: place it on the top (linter should throw error on it)

**Review by oriefrati — APPROVED** [review]

File: examples/client/Locomotion/android/app/src/main/java/com/locomotion/MainApplication.java (hunk 3 above)
Comment (lieblichaaron) [reply examples/client/Locomotion/android/app/src/main/java/com/locomotion/MainApplication.java]: it caused an error having it here and implementation project(':react-native-mixpanel') in app/build.gradle

File: examples/client/Locomotion/package.json (hunk 5 above)
Comment (lieblichaaron) [reply examples/client/Locomotion/package.json:23]: its native, i also have them in peer but i installed them as normal dependencies as well so i could work with the types while fixing things. does it need to be removed?

File: locomotion-sdk/src/Components/Drawer/index.js [hunk 7]
```diff
@@ -1,6 +1,9 @@
 import React from 'react';
-import { View } from 'react-native';
-import { DrawerItems, SafeAreaView } from 'react-navigation';
+import { SafeAreaView, View } from 'react-native';
```
Comment (OfryL) [inline locomotion-sdk/src/Components/Drawer/index.js:2]: Next time wrap `SafeAreaView` as component

File: locomotion-sdk/src/pages/Login/AuthLoadingScreen.js [hunk 8]
```diff
@@ -19,8 +19,8 @@ const AuthLoadingScreen = ({ navigation }) => {
         type: 'changeState',
         payload,
       });
-
-      let page = payload.userProfile ? 'App' : 'Auth';
+      console.log('payload', payload)
```
Comment (OfryL) [inline locomotion-sdk/src/pages/Login/AuthLoadingScreen.js]: remove

**Review by OfryL — APPROVED** [review]
check Mixpanel events after merge

**Pull request merged**